### ISSUE TITLE
Usage in NS-NG2 app

### ISSUE THREAD

**Issue by getmetorajesh** [PeterStaev/NativeScript-Grid-View#12]
Excellent plugin @PeterStaev, :clap:

Any sample on how I can use it in an NS-Angular2 application.

**Comment by minux84** [comment]
+1

**Comment by PeterStaev** [comment]
Hey guys, sadly I do not have much time to work on this plugin (now that Telerik has a ListView that allows a grid like view), but you can try to use it like all other vanilla {N} plugins (for example as explained here https://github.com/PeterStaev/NativeScript-Drop-Down#angular-2-example)

So something like 
```ts
// main.ts
import {registerElement} from "nativescript-angular/element-registry";
registerElement("GridView", () => require("nativescript-grid-view/grid-view").GridView);
```

And then use it in your HTML files by using the `<GridView/>`  and `<GridView.itemTemplate>` tags to configure the control and template respectively. So something like:
```xml
    <GridView>
      <GridView.itemTemplate>
        <GridLayout>
          <Label text="Text"/>
        </GridLayout>
      </GridView.itemTemplate>
    </GridView>
```

**Comment by PeterStaev** [comment]
After looking into this, seems what I suggested above does not work. So seems a special component wrapper must be implemented (similar to the `ListView` one [here](https://github.com/NativeScript/nativescript-angular/blob/master/nativescript-angular/directives/list-view-comp.ts)). Since I'm not very familiar with NG2 yet, I currently cannot help you on this. Feel free to implement and submit a PR, would gladly accept it 😄

**Comment by 98falcon** [comment]
any solution?

**Comment by m-abs** [comment]
I thought it looked like an interesting project to learn something, so I decided to try if I could solve it.

It's still very much a **WIP**: https://github.com/m-abs/nativescript-ngx-grid-view

**Comment by deepak4u2006** [comment]
I thing it can be achieved easily through, WrapLayout

`<ScrollView class="plansScroll" #plansScroll orientation="vertical" horizontalAlignment="left" width="100%">
    <WrapLayout orientation="horizontal" [id]="myIndex" (tap)="onPlanSectionTap($event)" class="plansScrollGrid m-5" #plansScrollGrid>
        <StackLayout *ngFor='let item of arrayMerchants; let myIndex = index' width="33.3%">
            <Image class="m-t-30 m-b-15" src="res://icon" width="80" height="80"></Image>
            <Label class="plansScrollGrid_Type m-5" minwidth="50" [text]='item.title' [color]='item.titleColor' horizontalAlignment="center" verticalAlignment="center" #plansScrollGridType textTransform="capitalize"></Label>
        </StackLayout>
    </WrapLayout>
</ScrollView>
`

**Comment by PeterStaev** [comment]
Great work @m-abs ! Feel free to submit a PR and I will gladly merge and provide any assistance if needed. And do not worry for the extra build step, I already have similar thing for the Drop Down and Masked Text Field widgets 😉

**Comment by m-abs** [comment]
@deepak4u2006 
From experience I wouldn't recommend it.
If you have only a few items, what you suggest would work fine, we have a similar view in our application.

But if you have too many items it will kill performance, because every item in your *ngFor will be rendered even, if they're not on screen at the moment.

It would be similar to using a ScrollView + a StackLayout + *ngFor in place of a ListView.

@PeterStaev 
Great, once I'm done, I'll integrate my code into your project. For now it's easier to keep it separate.

I noticed a few things working on this:
During development that if only templates are updated and livesynced, the gridview looks empty until the user interacts with the app, then it looks like they expand from the top-left corner.
I'm guessing something triggers the gridview into thinking item height/width should be zero.

I also noticed, if I navigate to another page and go back, the items are animated back into place. I'd expected them to be where they were before.
From the log output it looks like the gridview triggers some event while the its page is inactive.

Both cases are on my Android-6.0 emulator, I haven't tried it on iOS yet.
I don't know it they're unique to running in angular (and if so, if I caused it) or this behavior is also on pure nativescript?

**Comment by PeterStaev** [comment]
@m-abs , it really depends where do you load your data. If you are reloading your data in the `onNavigatedTo` event or `onLoaded` and you are not checking if it is from back navigation it is normal for the grid to redraw.

**Comment by deepak4u2006** [comment]
Hi @m-abs and @PeterStaev for the useful info, will surely take care of these things in my development.
@m-abs Mine case is displaying a number of menu items(static list), I think we both started at same time solving **Usage in NS-NG2 app** but I choose a shortcut due to some strict timelines.
I  am eagerly waiting for your WIP to come live. 👍 👍

**Comment by PeterStaev** [comment]
Thanks to @m-abs this now available in the 3.1 version of the plugin 👍 👏
